research-app/src/App.vue

```diff
@@ -1213,6 +1361,369 @@ export default class App extends WWTAwareComponent {
     }
   }
 
+  parseFloatParam(param: string | (string | null)[], fallback: number): number {
+    if (typeof param === 'string') {
+      let value = parseFloat(param);
+      return value || fallback;
+    }
+    return fallback;
+  }
+
+  encodeObjectBase64(obj: object): string {
+    return Buffer.from(JSON.stringify(obj)).toString('base64');
+  }
+
+  decodeObjectBase64(data: string): object {
+    return JSON.parse(Buffer.from(data, 'base64').toString());
+  }
+
+  adjustSettingsForImport(names: string[], values: any[]): void {
+    for (let i = 0; i < names.length; i++) {
+      const name = names[i];
+      const value = values[i];
+      if (name === 'color') {
```

Comment:
I don't *love* this special-casing ... this sort of specialization and de-specialization is the sort of thing that the "pywwt" types in research-app-messages handle. Is there a way to leverage that existing infrastructure?

Reply:
I was able to remove the special handling for `color` using the existing pywwt message conversion, but (unless I'm missing something) I don't see a way to do the same for the `Date`s. The issue is that the dates are being serialized as strings, but both spreadsheet layer setting types are expecting `Date` objects.

Depending on how we want users to pass in dates in messages (is a string fine, or do we really want a `Date` object), this code could be moved into the handler's `handleModifyMessage` method.